app/services/auth.server.ts

```diff
@@ -11,17 +12,17 @@ import {
 const url = `${
   process.env.VERCEL_URL
     ? `https://${process.env.VERCEL_URL}`
-    : "http://localhost:3000"
+    : `http://localhost:${process.env.PORT || 3000}`
 }/auth`;
 
 const github = new GitHubStrategy(
   {
     clientID: process.env.GH_CLIENT_ID as string,
     clientSecret: process.env.GH_CLIENT_SECRET as string,
-    callbackURL: `${url}/github/callback`,
+    callbackURL: `${url}${config.githubCallback}`,
```

Comment:
`path.join()`  is great for this prevents having accidental too little or too many slashes